apps/api/pages/api/teams/_post.ts

```diff
@@ -49,8 +76,20 @@ import { schemaTeamBodyParams, schemaTeamReadPublic } from "~/lib/validations/te
  *        description: Authorization information is missing or invalid.
  */
 async function postHandler(req: NextApiRequest) {
-  const { prisma, body, userId } = req;
-  const data = schemaTeamBodyParams.parse(body);
+  if (!IS_TEAM_BILLING_ENABLED) {
+    const PRECONDITION_FAILED_STATUS_CODE = 412;
+    throw new HttpError({
+      statusCode: PRECONDITION_FAILED_STATUS_CODE,
+      message: "Team creation is unavailable because team billing is disabled.",
+    });
+  }
```

Comment:
Will this prevent team creation for self-hosters that don't use team billing? If so, we should just create the team normally if billing is not enabled.

Reply:
Good point. Done in [Create teams as before if billing disabled](https://github.com/calcom/cal.com/pull/13211/commits/5f0381ce83e5058edb4b45f284b99b005b3cc0ce)